### PR TITLE
test: add comprehensive edge cases for IsUUID validation

### LINKED ISSUE #17818: [mesheryctl] Add comprehensive unit tests for IsUUID validation
## Description
The current `IsUUID` utility in `mesheryctl` has basic test coverage, but it lacks verification for several valid UUID formats and common CLI input edge cases. 

To ensure the robustness of the utility, this issue tracks the addition of unit tests for:
- Braced UUID format `{...}`
- URN UUID format `urn:uuid:...`
- Input strings with leading/trailing whitespace
- Excessively long input strings
- Empty string handling

## Acceptance Criteria
- [ ] Implement table-driven tests in `validator_test.go`.
- [ ] Use the "Given, When, Then" naming convention for test cases.
- [ ] Verify all edge cases pass successfully in the CI pipeline.

### PR DESCRIPTION
**Pull request by Mustansir-06**
This PR enhances the unit test suite for the IsUUID utility in mesheryctl. While the existing tests covered standard UUIDs, they lacked coverage for alternative valid formats and common string-handling edge cases encountered in CLI inputs.

Changes:

Added Valid Edge Cases: Included support for Braced format {...} and URN prefix urn:uuid:..., both of which are valid UUID representations supported by the underlying google/uuid package.

Added Invalid Edge Cases: Introduced tests for leading and trailing whitespace, as well as strings that exceed the maximum valid UUID length.

Improved Test Runner Logic: * Refactored the sub-test naming logic to provide descriptive labels for "empty string" and "whitespace" inputs in the test output.

Added a safety check to prevent index-out-of-range errors during sub-test naming.

Fixes #17818

### CODE REVIEW

**Comment by shri771** [comment]
@Mustansir-06, could you resolve the failing lint check?

File: mesheryctl/pkg/utils/validator_test.go [hunk 1]
```diff
@@ -28,14 +32,23 @@ func TestIsUUID(t *testing.T) {
...
-		// if name == "" {
-		// 	name = "empty"
-		// }
-		tc := tc
+		if name == "" {
+			name = "empty string"
+		} else if strings.TrimSpace(name) != name { 
+			name = "whitespace edge case: '" + tc.in + "'"
+		}
+
```
Comment (lekaf974) [inline mesheryctl/pkg/utils/validator_test.go]: this is just confusing please add a field name in struct that describe the expected behavior based on input.

we introduce given, when, then naming convention in tests recently

**Review by lekaf974 — CHANGES_REQUESTED** [review]
Please create an issue for this work and ping me to assign the issue to you and fix the comment

File: mesheryctl/pkg/utils/validator_test.go
Comment (lekaf974) [inline mesheryctl/pkg/utils/validator_test.go]:
```suggestion
		name string
```
comment is useless

File: mesheryctl/pkg/utils/validator_test.go [hunk 2]
```diff
@@ -9,37 +9,67 @@ import (
...
 
-	for _, tc := range cases {
-		name := tc.in
-		// if name == "" {
-		// 	name = "empty"
-		// }
-		tc := tc
-		t.Run(name, func(t *testing.T) {
+	for _, tc := range tests {
+		tc := tc // Capture range variable
```
Comment (lekaf974) [inline mesheryctl/pkg/utils/validator_test.go:68]: is this really necessary?

Comment (lekaf974) [reply mesheryctl/pkg/utils/validator_test.go:68]: @Mustansir-06 any update ?
